psp34_shiden/contracts/shiden34/src/lib.rs

```diff
@@ -63,7 +69,37 @@ pub mod shiden34 {
         #[ink(message)]
         fn max_supply(&self) -> u64;
         #[ink(message)]
+        fn price(&self) -> Balance;
```

Comment:
`mint_price`? otherwise, it's not clear what kind of price, it could be token price.